src/lib/utils/package-manifest.ts

```diff
@@ -0,0 +1,193 @@
+// Utilities to support the inspection of node package "manifests" (package.json's)
+
+import { existsSync, statSync } from "fs";
+import glob = require("glob");
+import { dirname, resolve } from "path";
+import { flatMap } from "./array";
+
+import { readFile } from "./fs";
+import { Logger } from "./loggers";
+
+/**
+ * Helper for the TS type system to understand hasOwnProperty
+ * and narrow a type appropriately.
+ * @param obj the receiver of the hasOwnProperty method call
+ * @param prop the property to test for
+ */
+function hasOwnProperty<X extends {}, Y extends PropertyKey>(
```

Comment:
This is a very generic utility function but I wasn't sure if it should live elsewhere.

Reply:
Here is fine, if something else needs it it can always be moved.